### PR TITLE
implement course search component

### PR DESCRIPTION
**Pull request by m-duran**
Creates course search component. Currently the service fetches all of the courses and filters them afterwards. The api needs to be extended to take a search term so the service call use that instead. 

The changes in app.module.ts, other than adding the new components and services, are just reordering the existing ones in alphabetical order.

### CODE REVIEW

**Comment by m-duran** [comment]
#97

File: webapp/app/components/course-search/course-search.component.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        private courseSearchService: CourseSearchService,
+        private router: Router) {}
+
+    search(term: string): void {
+        this.searchTerms.next(term);
+    }
+
+    ngOnInit(): void {
+        this.courses = this.searchTerms
+            .debounceTime(300)
```
Comment (macrivello) [inline webapp/app/components/course-search/course-search.component.ts]: how awesome is this?!

File: webapp/app/components/flowchart/flowchart.component.html [hunk 2]
```diff
@@ -18,3 +18,5 @@
     </ng-template>
   </div>
 </div>
+
+<course-search></course-search>
```
Comment (macrivello) [inline webapp/app/components/flowchart/flowchart.component.html]: I think we want this displayed in a modal. 

Check out: 
https://material.angular.io/components/component/dialog

Comment (m-duran) [inline webapp/app/components/flowchart/flowchart.component.html]: Yea I just put it there for testing purposes, that line can be removed. 

Thanks for the link, I was looking for something like that.

**Pull request merged**